torontoadvotech/client
# Ws 17 - Email Validation Page

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description


Added `ValidateEmail` page and updated code to check if email is validated. Upon initial load, we make a call to the validate email endpoint of the API and pass in the ID of the user to update (`PATCH`) that their email is validated. If successful, the user will see "Thank you for verifying your email” message. Otherwise, they will see "Something went wrong, please try again".

## Related Issue


https://torontoadvotech.atlassian.net/jira/software/projects/WS/boards/1?selectedIssue=WS-17

## Acceptance Criteria


- This page should send a request to the validateEmail route of the API with the user’s ID as soon as the user lands on the page. If the request is successful the user should see a “Thank you for validating” page. Otherwise they should see an error page.

- Render a blank page with a black background and a h1 text in the primary brand colour

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before


N/A

### After


![Screen Shot 2021-03-05 at 9 28 21 PM](https://user-images.githubusercontent.com/46036289/110192036-c75cab80-7df9-11eb-920b-f5c1751cdfee.png)

## Testing Steps / QA Criteria


1. Pull from branch `WS-17`
2. In Postman, send a `POST` Sign Up request and copy the `id` that is generated. 
3. In Postman, send a `PATCH` Verify Email request and ensure your id is used in place of `:userId` and check that the status is successful 
4. In your localhost navigate to validate email page ie: http://localhost:8888/validateEmail/6042d3c12fc1c207f5786def and update the URL with your id (`http://localhost:8888/validateEmail/[id]`). You should see "Thank you for verifying your email." if your patch request was successful. 
5. Change the URL to use an invalid id and you should see "Something went wrong, please try again."
